### ISSUE TITLE
SSLProtocolException when downloading tiles

### ISSUE THREAD

**Issue by Legista** [osmdroid/osmdroid#1525]
## Issue Type

[x] Question
[x] Bug

## Description

A lot of errors like this:
```
W/OsmDroid: IOException downloading MapTile: /7/75/42 : javax.net.ssl.SSLProtocolException: Read error: ssl=0xb307b980: Failure in SSL library, usually a protocol error
    error:1e06e065:Cipher functions:aead_aes_gcm_open:BAD_DECRYPT (external/boringssl/src/crypto/cipher/e_aes.c:1188 0xaac6ad97:0x00000000)

```
And some tiles are not downloading.

There weren't such errors a week ago.

## Steps to reproduce the problem
Tile source - MAPNIK
Zoom - 5 and more.
For example, when OpenTopo everything is ok.

### If it's a bug, version(s) of android this affects:

Tested on API 23 and 29.

### Version of osmdroid the issue relates to:

6.1.5

**Comment by monsieurtanuki** [comment]
> There weren't such errors a week ago.

If you didn't change your code and your version of osmdroid, I guess this problem comes from the source, not from osmdroid.

Or perhaps it's related to "[I cannot see the map tiles on my Android 9+](https://github.com/osmdroid/osmdroid/wiki/FAQ#i-cannot-see-the-map-tiles-on-my-android-9)"

Anyway, I've just tested with the demo app: I have no issue with MAPNIK tiles.
Is it systematic or just on some random tiles?

**Comment by Legista** [comment]
I didn't change my code and version of osmdroid.

> Is it systematic or just on some random tiles?
There is endless stream of this errors in Logcat when zoom is 5 or more.

> Or perhaps it's related to "[I cannot see the map tiles on my Android 9+](https://github.com/osmdroid/osmdroid/wiki/FAQ#i-cannot-see-the-map-tiles-on-my-android-9)"
No, it's not issue, cause i set up network security and my application was ran on API 29 as like on 23.

> I guess this problem comes from the source, not from osmdroid.
It was my guess too, but it's strange, that only i have this errors.

**Comment by monsieurtanuki** [comment]
Well, I've tried and on my smartphone everything's fine with MAPNIK. The best I can suggest is to try with the latest osmdroid release.

**Comment by Legista** [comment]
Please, help me!
What i'm doing wrong and what else can i do?
Here is video, where i am creating new test app from scratch (with latest osmdroid release). 
https://youtu.be/HuGQsgRLSxo

**Comment by Legista** [comment]
I have builded test app with older release of OSMDroid (5.6.5).
And got another errors, like this:
`W/OsmDroid: IOException downloading MapTile: /6/39/20 : java.net.ProtocolException: unexpected end of stream`

**Comment by monsieurtanuki** [comment]
Correct me if I'm wrong, but in your video I see that the map looks as if it were correctly downloaded - which is good enough for me - but sometimes you see logs that you don't like.

**Comment by Legista** [comment]
> Correct me if I'm wrong
A while back i noticed tiles downloading slowly, i looked in logcat and thought those errors are not good.
So i went here and just hoped someone answered me like this way "hey, i have this errors too, it's normal, just drop it". That would be enough for me to stop worrying ))

**Comment by Legista** [comment]
> I see that the map looks as if it were correctly downloaded

Actually... It's not entirely true.
https://youtu.be/ECeGcr6dnuQ

**Comment by monsieurtanuki** [comment]
Let me rephrase the issue: a tiny minority of tiles from MAPNIK have trouble being downloaded in osmdroid.
A recent example being http://a.tile.openstreetmap.org/6/39/20.png, which can be reached without problem by my internet browser, but throws an Exception is osmdroid:
````
W/OsmDroid: IOException downloading MapTile: /6/39/20 : java.net.ProtocolException: unexpected end of stream
````

**Comment by Legista** [comment]
This error from older releases and in case of using custom source wthout httpS.
In latest release - throws SSLProtocolException.
I can access any reported tile from my internet browser too.
Some of the tiles load after all.
Some of them - not, like in video above (https://youtu.be/ECeGcr6dnuQ).

**Comment by monsieurtanuki** [comment]
I'm not saying that's OK; I just want to know what we're talking about.
From your OP I imagined that nothing at all was visible from MAPNIK from level 5 (something similar happened around the http/httpS/API29 situation).

The reality is much less dramatic.

I couldn't reproduce the issue and succesfully downloaded the tile around Algiers/Médéa (cf. your video) from MAPNIK https.
And I couldn't reproduce either your issue with the http tile:
````java
new Thread(new Runnable() {
	@Override
	public void run() {
		final SqlTileWriter sqlTileWriter = new SqlTileWriter();
		final TileDownloader tileDownloader = new TileDownloader();
		final OnlineTileSourceBase testSource = new XYTileSource("Mapnik",
				0, 19, 256, ".png", new String[] {
				"http://a.tile.openstreetmap.org/" },"© OpenStreetMap contributors",
				new TileSourcePolicy(2,
						TileSourcePolicy.FLAG_NO_BULK
								| TileSourcePolicy.FLAG_NO_PREVENTIVE
								| TileSourcePolicy.FLAG_USER_AGENT_MEANINGFUL
								| TileSourcePolicy.FLAG_USER_AGENT_NORMALIZED
				));
		try {
			final Drawable drawable = tileDownloader.downloadTile(MapTileIndex.getTileIndex(6, 39, 20), sqlTileWriter, testSource);
			Log.e("TEST", "done/" + drawable);
		} catch(CantContinueException e) {
			//
		}
	}
}).start();
````

**Comment by Legista** [comment]
It's intresting.
I put this code in button's onClick and pressed button several times.
Here is result
https://i.imgur.com/HjBq8Sx.png
https://i.imgur.com/iSG5Iuy.png

**Comment by Legista** [comment]
Oh damn!!! It's something wrong with emulator!
So stupid, i have to did this before! (facepalm)
I just opened tile link in emulator's browser and "voilà" - sometimes it's loaded but sometimes:
https://i.imgur.com/aTWPJLs.png

It happens with any virtual device.

**Comment by monsieurtanuki** [comment]
I guess we cannot expect osmdroid to download png files that even browsers cannot read, can we?
@Legista Unless there's something to add I think it makes sense to close this issue.

**Comment by Legista** [comment]
> I guess we cannot expect osmdroid to download png files that even browsers cannot read, can we?

Shure :)))
